include/mbedtls/md5.h

```diff
@@ -36,6 +36,11 @@
 // Regular implementation
 //
 
+#if ( defined(__ARMCC_VERSION) || defined(_MSC_VER) ) && \
```

Comment:
Not sure why do we add this definition in many separate headers (currently 9) instead of making a shared one for it, but maybe it would be an overkill.

Reply:
See https://github.com/ARMmbed/mbedtls/pull/1294#issuecomment-360094307